nebula_ros/src/continental/continental_ars548_hw_interface_ros_wrapper.cpp

```diff
@@ -0,0 +1,329 @@
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("configuration_sensor_port", 2368, descriptor);
+    sensor_configuration.configuration_sensor_port =
+      this->get_parameter("configuration_sensor_port").as_int();
+  }
+
+  if (sensor_configuration.sensor_model == nebula::drivers::SensorModel::UNKNOWN) {
+    return Status::INVALID_SENSOR_MODEL;
+  }
+
+  if (sensor_configuration.frame_id.empty()) {
+    return Status::SENSOR_CONFIG_ERROR;
+  }
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "SensorConfig:" << sensor_configuration);
+  return Status::OK;
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::ReceivePacketsDataCallback(
+  std::unique_ptr<nebula_msgs::msg::NebulaPackets> scan_buffer)
+{
+  packets_pub_->publish(std::move(scan_buffer));
+}
+
+rcl_interfaces::msg::SetParametersResult ContinentalARS548HwInterfaceRosWrapper::paramCallback(
+  const std::vector<rclcpp::Parameter> & p)
+{
+  std::scoped_lock lock(mtx_config_);
+  RCLCPP_DEBUG_STREAM(this->get_logger(), "add_on_set_parameters_callback");
+  RCLCPP_DEBUG_STREAM(this->get_logger(), p);
+  RCLCPP_DEBUG_STREAM(this->get_logger(), sensor_configuration_);
+  RCLCPP_INFO_STREAM(this->get_logger(), p);
+
+  drivers::ContinentalARS548SensorConfiguration new_param{sensor_configuration_};
+  RCLCPP_INFO_STREAM(this->get_logger(), new_param);
+  std::string sensor_model_str;
+  std::string return_mode_str;
+  if (
+    get_param(p, "sensor_model", sensor_model_str) ||
+    get_param(p, "return_mode", return_mode_str) || get_param(p, "host_ip", new_param.host_ip) ||
+    get_param(p, "sensor_ip", new_param.sensor_ip) ||
+    get_param(p, "frame_id", new_param.frame_id) ||
+    get_param(p, "data_port", new_param.data_port) ||
+    get_param(p, "configuration_host_port", new_param.configuration_host_port) ||
+    get_param(p, "configuration_sensor_port", new_param.configuration_sensor_port)) {
+    if (0 < sensor_model_str.length())
+      new_param.sensor_model = nebula::drivers::SensorModelFromString(sensor_model_str);
+
+    sensor_configuration_ = new_param;
+    RCLCPP_INFO_STREAM(this->get_logger(), "Update sensor_configuration");
+    std::shared_ptr<drivers::SensorConfigurationBase> sensor_cfg_ptr =
+      std::make_shared<drivers::ContinentalARS548SensorConfiguration>(sensor_configuration_);
+    RCLCPP_DEBUG_STREAM(this->get_logger(), "hw_interface_.SetSensorConfiguration");
+    hw_interface_.SetSensorConfiguration(
+      std::static_pointer_cast<drivers::SensorConfigurationBase>(sensor_cfg_ptr));
+    hw_interface_.CheckAndSetConfig();
+  }
+
+  auto result = std::make_shared<rcl_interfaces::msg::SetParametersResult>();
+  result->successful = true;
+  result->reason = "success";
+
+  RCLCPP_DEBUG_STREAM(this->get_logger(), "add_on_set_parameters_callback success");
+
+  return *result;
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::DrivingDirectionCallback(
+  [[maybe_unused]] const std_msgs::msg::Bool::SharedPtr msg)
+{
```

Comment:
Outputting a warning (e.g. 'Ignored received odometry: callback not yet implemented') to the user when trying to publish to this topic would be helpful.

Reply:
@mojomex 
With respect to the comments in the ros_wrappers, how to use them both generally in ros or autoware has not been discussed yet (hence the reason this PR is still a draft).

I will address this comments later when we have a better idea of how to handle all of our use-cases